components/uniset/src/conversions.rs

```diff
@@ -0,0 +1,123 @@
+use crate::utils::deconstruct_range;
+use crate::UnicodeSet;
+use std::{
+    convert::TryFrom,
+    ops::{Range, RangeBounds, RangeFrom, RangeFull, RangeInclusive, RangeTo, RangeToInclusive},
+};
+
+fn try_from_range_impl(range: impl RangeBounds<char>) -> Result<UnicodeSet, (u32, u32)> {
+    let (from, till) = deconstruct_range(range);
+    if from < till {
+        let set = vec![from, till];
+        Ok(UnicodeSet::try_from(set).unwrap())
+    } else {
+        Err((from, till))
+    }
+}
+
+impl TryFrom<Range<char>> for UnicodeSet {
+    type Error = String;
+
+    fn try_from(range: Range<char>) -> Result<Self, Self::Error> {
+        match try_from_range_impl(range) {
```

Comment:
`try_from_range_impl(range)?`

Reply:
Do you have a recommendation for a better naming scheme? Since all the conversions require the `try_from_range_impl`, I kept the naming scheme the same as `contains` and `contains_impl`